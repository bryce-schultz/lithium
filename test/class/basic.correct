0
0
{ x: 0, y: 0 }
10
20
{ x: 10, y: 20 }
