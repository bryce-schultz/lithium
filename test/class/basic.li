import <io>

class pair
{
    let x = 0;
    let y = 0;

    pub fn pair(x, y)
    {
        this.x = x;
        this.y = y;
    }

    pub fn getX()
    {
        return this.x;
    }

    pub fn getY()
    {
        return this.y;
    }
}

let p = pair(10, 20);
println(p.getX());
println(p.getY());
