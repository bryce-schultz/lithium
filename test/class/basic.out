error: basic.li:1:1: unexpected token
│ class pair
│ ~~~~~
│ ^
